internal/repositories/hash_storage.go

```diff
@@ -0,0 +1,76 @@
+package repositories
+
+import (
+	"database/sql"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/models"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/api"
+	"github.com/sirupsen/logrus"
+)
+
+type HashStorage struct {
+	db     *sql.DB
+	alg    string
+	logger *logrus.Logger
+}
+
+// NewHashStorage creates a new struct HashService
+func NewHashStorage(db *sql.DB, alg string, logger *logrus.Logger) *HashStorage {
+	return &HashStorage{
+		db:     db,
+		alg:    alg,
+		logger: logger,
+	}
+}
+
+// Create accesses the repository to save data to the database
+func (hs HashStorage) Create(allHashData []*api.HashData, deploymentData *models.DeploymentData) error {
+	tx, err := hs.db.Begin()
```

Comment:
Please find the `WithTx` function in our code. You may use it to run some code with a DB transaction.

Reply:
resolved and used WithTx func  https://github.com/ScienceSoft-Inc/integrity-sum/commit/df701c8e8df4c9c3625d41f08958dcbab2fcd4c6